scripts/manage-db.py

```diff
@@ -123,64 +123,79 @@ def extract_active_data(trans, url, dump_location='dump.sql'):
-    )
-
-    popen(
-        _strip_multiple_spaces('%s --skip-add-drop-table --no-create-info %s releases_history --where "name = ( \
-            SELECT rules.mapping \
-            FROM rules \
-            WHERE rules.alias=\'firefox-release\' \
-        ) LIMIT 50" >> %s' % (mysql_default_command, database, dump_location))
+        _strip_multiple_spaces("""
+            %s --skip-add-drop-table --no-create-info %s
+            dockerflow rules rules_history migrate_version >> %s
+        """ % (mysql_default_command, database, dump_location)
+        )
     )
 
+    # Because Releases are so massive, we only want the actively used ones,
+    # and very little Release history. Specifically:
+    #   - All releases referenced by a Rule or a Scheduled Rule Change
+    #   - All releases referenced by a Release from the above query
+    #   - 50 rows of history for the "Firefox-mozilla-central-nightly-latest" Release
+    #   - Full history for the Release currently referenced by the "firefox-release" Rule.
     query_release_mapping = """SELECT DISTINCT releases.* \
         FROM releases, rules, rules_scheduled_changes \
-        WHERE releases.name IN ( \
+        WHERE rules_scheduled_changes.complete = 0 AND releases.name IN ( \
             rules.mapping, \
             rules.fallbackMapping, \
             rules_scheduled_changes.base_mapping, \
             rules_scheduled_changes.base_fallbackMapping \
             )"""
 
     result = trans.execute(query_release_mapping).fetchall()
-    partial_release_names = set()
     release_names = set()
     for row in result:
         try:
             release_names.add(str(row['name']))
             release_blob = createBlob(row['data'])
-            partial_release_names.update(release_blob.getReferencedReleases())
+            release_names.update(release_blob.getReferencedReleases())
         except ValueError:
             continue
-    partial_release_names.difference_update(release_names)
-    if partial_release_names:
-        batch_generator = chunk_list(list(partial_release_names), 30)
-        for batched_partial_release_list in batch_generator:
-            qry = ", ".join("'" + release_names + "'" for release_names in batched_partial_release_list)
+    if release_names:
+        batch_generator = chunk_list(list(release_names), 30)
+        for batched_release_list in batch_generator:
+            qry = ", ".join("'" + names + "'" for names in batched_release_list)
             popen(_strip_multiple_spaces('%s --skip-add-drop-table --no-create-info %s '
                                          'releases --where="releases.name IN (%s)" \
                    >> %s' % (mysql_default_command, database, qry, dump_location)))
 
+    popen(
+        _strip_multiple_spaces('%s --skip-add-drop-table --no-create-info %s releases_history --where="releases_history.name=\'Firefox-mozilla-central-nightly-latest\' \
+        limit 50" >> %s' % (mysql_default_command, database, dump_location))
```

Comment:
I see you've only mobved this, but does it grab the first 50 history rows or last 50 ? aka might need an ORDER clause too. Same for the next query.

Reply:
I'll double check this. I imagine we want the 50 most recent ones...